integration_tests/src/main/python/subquery_test.py

```diff
@@ -63,45 +68,94 @@ def test_scalar_subquery_struct(basic_gen):
         where (select first(ss) from table).arr[0] > ss.arr[1]
         ''')
 
+
 @ignore_order(local=True)
+@pytest.mark.parametrize('is_ansi_enabled', [False, True])
 @pytest.mark.parametrize('basic_gen', all_basic_gens, ids=idfn)
-def test_scalar_subquery_array(basic_gen):
+def test_scalar_subquery_array(is_ansi_enabled, basic_gen):
+    """
+    For this test, all the array inputs are sized so that ArrayIndexOutOfBounds conditions are
+    avoided.  This is to ensure that the tests don't fail with exceptions in ANSI mode.
+    Note that no meaningful test coverage is lost here.  ArrayIndexOutOfBounds exceptions are
+    already tested as part of array_test.py::test_array_item_ansi_fail_invalid_index.
+    """
+    conf = {'spark.sql.ansi.enabled': is_ansi_enabled}
+
     # single-level array
+    test_array_gen = ArrayGen(basic_gen, min_length=1 if is_ansi_enabled else 0)
     assert_gpu_and_cpu_are_equal_sql(
         # Fix num_slices at 1 to make sure that first/last returns same results under CPU and GPU.
-        lambda spark: gen_df(spark, [('arr', ArrayGen(basic_gen))], num_slices=1),
-        'table',
+        lambda spark: gen_df(spark, [('arr', test_array_gen)], num_slices=1),
+        'my_array_table',
```

Comment:
Sorry, missed this in the prior review.  We should never be making up table names manually.  There's a generator for unique table names, see spark_tmp_table_factory and how it's used in other tests.  Table names in tests should _always_ be generated from that.

Reply:
Thank you, yes.  This entire test-script has this going on.

I'll make the correction.  I think the entire file might benefit from that change.

Reply:
Fixed.  Made the change for all the tests in the file.